src/witan/send/adroddiad/dsg_plan.clj

```diff
@@ -0,0 +1,174 @@
+(ns witan.send.adroddiad.dsg-plan
+  (:require [com.climate.claypoole.lazy :as lazy]
+            [tablecloth.api :as tc]
+            [tech.v3.dataset.reductions :as ds-reduce]
+            [tech.v3.datatype.functional :as dfn]
+            [witan.send.adroddiad.summary.report :as summary-report]))
+
+;; Dataset needs to be in a canonical census with transition-count shape for this to work
+(defn ncy->age
+  "Taken from witan.send.domain.academic-years to avoid a dependency"
+  [ncy]
+  (+ ncy 5))
+
+(defn age->ncy
+  "Taken from witan.send.domain.academic-years to avoid a dependency"
+  [age]
+  (- age 5))
+
+(defn age-group
+  "Taken from witan.send.domain.academic-years to avoid a dependency"
+  [age]
+  (cond
+    (< age 5) "Age 0 to 5"
+    (<= 5 age 10) "Age 05 to 10"
+    (<= 11 age 15) "Age 11 to 15"
+    (<= 16 age 19) "Age 16 to 19"
+    (<= 20 age 25) "Age 20 to 25"
+    (< 25 age) "Over 25"))
+
+(defn summarise-setting-by-age [census]
+  (-> census
+      (tc/map-columns :age-group [:academic-year]
+                      (fn [ay]
+                        (-> ay
+                            ncy->age
+                            age-group)))
+      (tc/group-by [:calendar-year :setting :age-group])
+      (tc/aggregate {:transition-count #(dfn/sum (:transition-count %))})
+      (tc/complete :calendar-year :setting :age-group)
+      (tc/replace-missing :transition-count :value 0)))
+
+(defn summarise-setting-by-need [census]
+  (-> census
+      (tc/group-by [:calendar-year :setting :need])
+      (tc/aggregate {:transition-count #(dfn/sum (:transition-count %))})
+      (tc/complete :calendar-year :setting :need)
+      (tc/replace-missing :transition-count :value 0)))
+
+(defn transform-simulations
+  [{:keys [simulations
+           simulations-transform-f
+           cpu-pool]
+    :or {cpu-pool (java.util.concurrent.ForkJoinPool/commonPool)}}]
+  (lazy/upmap cpu-pool simulations-transform-f simulations))
+
+(defn summarise-simulations [{:keys [simulations
```

Comment:
For other functions here e.g. `summarise-setting-by-need`, `value-key` is not parameterised. Why parameterise here and not in the other places?

Reply:
the simulation-transform might change the underlying key that we are working on, whereas `summarise-by-*` is focused just on the problem at hand. I could probably remove the :transition-count tweak, but I think it is OK for now.